cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -197,26 +198,27 @@ func generateInitramfsMounts() (err error) {
 
 // generateMountsMode* is called multiple times from initramfs until it
 // no longer generates more mount points and just returns an empty output.
-func generateMountsModeInstall(mst *initramfsMountsState) error {
+func generateMountsModeInstall(mst *initramfsMountsState) (gadget.Model, error) {
 	// steps 1 and 2 are shared with recover mode
 	model, snaps, err := generateMountsCommonInstallRecover(mst)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	// 3. final step: write modeenv to tmpfs data dir and disable cloud-init in
 	//   install mode
 	modeEnv, err := mst.EphemeralModeenvForModel(model, snaps)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	if err := modeEnv.WriteTo(boot.InitramfsWritableDir); err != nil {
-		return err
+	isRunMode := false
+	if err := modeEnv.WriteTo(boot.InitramfsWritableDir(model, isRunMode)); err != nil {
+		return nil, err
 	}
 
 	// done, no output, no error indicates to initramfs we are done with
 	// mounting stuff
-	return nil
+	return model, nil
```

Comment:
What about making a separate PR where this returns `nil, nil`? It would consist of most of the boring parts of the diff, and then reviewing the rest should be easier.

Reply:
Great point - with Samueles idea to have a "mst.verifiedModel" this would also be less churn.